contracts/test-utils/src/vote_factory.ts

```diff
@@ -0,0 +1,79 @@
+import { assert } from '@0x/assert';
+import { schemas } from '@0x/json-schemas';
+import { eip712Utils } from '@0x/order-utils';
+import { ECSignature, EIP712DomainWithDefaultSchema, EIP712TypedData, SignatureType } from '@0x/types';
+import { BigNumber, signTypedDataUtils } from '@0x/utils';
+import * as ethUtil from 'ethereumjs-util';
+import * as _ from 'lodash';
+
+import { constants } from './constants';
+import { signingUtils } from './signing_utils';
+import { SignedZeroExVote, ZeroExVote } from './types';
+
+export class VoteFactory {
+    private readonly _delegatorPrivateKey: Buffer;
+    private readonly _domain: EIP712DomainWithDefaultSchema;
+
+    public static newZeroExVote(
+        proposalId: BigNumber,
+        support: boolean,
+        operatedPoolIds: number[],
+    ): ZeroExVote {
+        return {
+            proposalId,
+            support,
+            operatedPoolIds,
+        };
+    }
+
+    constructor(
+        delegatorPrivateKey: Buffer,
+        contractName: string,
+        chainId: number,
+        verifyingContract: string,
+    ) {
+        this._delegatorPrivateKey = delegatorPrivateKey;
+        this._domain = {name: contractName, chainId, verifyingContract};
+    }
+
+    public async newSignedZeroExVoteAsync(
+        vote: ZeroExVote,
+        signatureType: SignatureType = SignatureType.EIP712,
+    ): Promise<SignedZeroExVote> {
+        const voteHashBuffer = this._getVoteHashBuffer(vote);
+        const signature = signingUtils.signMessage(voteHashBuffer, this._delegatorPrivateKey, signatureType);
+        return {
+            ...vote,
+            signature: `0x${signature.toString('hex')}`,
+        };
+    }
+
+    private _getVoteHashBuffer(vote: ZeroExVote | SignedZeroExVote): Buffer {
+        const typedData = this._createZeroExVoteTypedData(vote);
+        return signTypedDataUtils.generateTypedDataHash(typedData);
+    }
+
+    private _createZeroExVoteTypedData(zeroExVote: ZeroExVote): EIP712TypedData {
+        assert.doesConformToSchema('ZeroExVote', zeroExVote, schemas.ZeroExVote); // TODO(need to bump @0x/json-schemas first?)
+        const normalizedVote = _.mapValues(zeroExVote, value => {
+            return !_.isString(value) ? value.toString() : value;
+        });
+        return eip712Utils.createTypedData(
+            // TODO(need to bump @0x/json-schemas first?)
+            constants.ZEROEX_VOTE_SCHEMA.name,
+            { ZeroExVote: constants.ZEROEX_VOTE_SCHEMA.parameters },
+            normalizedVote,
+            this._domain,
+        );
+    }
```

Comment:
Hi @Recmo , can you help confirm that for this test, I will need to define/update in 0x/tools/json-schemas for this? 🙏

Reply:
I'd say let's move this stuff to `@0x/contracts-treasury`. There's a chance the website will use this tooling, so best to keep everything together I think.
Also I personally prefer the abstractions used [here](https://github.com/0xProject/protocol/blob/development/packages/protocol-utils/src/orders.ts) for EIP712 stuff.

Reply:
Thanks Michael, I have moved these to `contracts/treasury/src/votes.ts`, not really confident about the code structure, would be great to get your opinion on that. thx!